packages/manager/src/types.ts

```diff
@@ -0,0 +1,59 @@
+import { HookError, SliceMachinePluginOptions } from "@slicemachine/plugin-kit";
+import { VERSION_KIND } from "./constants/VERSION_KIND";
+
+export type PackageManager = "npm" | "yarn";
+
+export type PackageChangelog = {
+	currentVersion: string;
+	updateAvailable: boolean;
+	latestNonBreakingVersion: string | null;
+	versions: PackageVersion[];
+};
+
+export type PackageVersion = {
+	versionNumber: string;
+	releaseNote: string | null;
+	kind: typeof VERSION_KIND[keyof typeof VERSION_KIND] | null;
+};
+
+/**
+ * A string, object, or instance representing a registered plugin.
+ *
+ * @typeParam TSliceMachinePluginOptions - User-provided options for the plugin.
+ */
+export type SliceMachineConfigPluginRegistration<
+	TSliceMachinePluginOptions extends SliceMachinePluginOptions = SliceMachinePluginOptions,
+> =
+	| string
+	| {
+			resolve: string;
+			options?: TSliceMachinePluginOptions;
+	  };
+
+/**
+ * Slice Machine configuration from `slicemachine.config.js`.
+ */
+export type SliceMachineConfig = {
+	// TODO: Is `_latest` necessary? Can we deprecate it?
```

Comment:
when slicemachine is installed `_latest` gets set o the current version.

Reply:
I think we should re-think how migrations are done today. `_latest` is a stateful property, and `sm.json` is not for state; it's for user configuration.

We could detect if a project needs a migration each start up rather than checking `_latest`. That prevents the case where `_latest` is the latest version, but a migration is needed.